app/sandbox/scoring.py

```diff
@@ -0,0 +1,142 @@
+from datetime import datetime, timezone
+
+from .models import EnglishLevel, JobPosting, Message
+from .score_algorithm import ScoreAlgorithm, LogicalRule, SimpleRule, IterableRule
+
+ENGLISH_WEIGTH = {
+    EnglishLevel.NONE: 0,
+    EnglishLevel.BASIC: 0.2,
+    EnglishLevel.PRE: 0.5,
+    EnglishLevel.INTERMEDIATE: 1,
+    EnglishLevel.UPPER: 2,
+    EnglishLevel.FLUENT: 3
+}
+
+EXPERIENCE_WEIGTH = {
+    JobPosting.Experience.ZERO: 0,
+    JobPosting.Experience.ONE: 1,
+    JobPosting.Experience.TWO: 2,
+    JobPosting.Experience.THREE: 3,
+    JobPosting.Experience.FIVE: 4
+}
+
+DATE_WEIGTH = 15
+
+
+def score_thread(thread):
+    scoring = ScoreAlgorithm()
+
+    candidate = thread.candidate
+    job = thread.job
+
+    candidate_full_info = ((candidate.moreinfo or '') + ' ' +
+                           (candidate.looking_for or '') + ' ' +
+                           (candidate.highlights or ''))
+    clear_candidate_full_info = clear_text(candidate_full_info.lower())
+
+    clear_job_description = clear_text(job.long_description.lower()).split()
+    job_candidate_desc_occurs = set(clear_candidate_full_info.split()).intersection(clear_job_description)
+
+    scoring.add(LogicalRule(score=1,
+                            when=(datetime.now(timezone.utc) - thread.last_updated).days < 7
+                                 and thread.last_message.sender == Message.Sender.CANDIDATE,
+                            description="Increase score if candidate respond within last 7 days"))
+
+    scoring.add(LogicalRule(score=0.2,
+                            when=candidate.primary_keyword.lower() in job.primary_keyword.lower(),
+                            description='Candidate primary keyword must exist in job primary keyword'))
+
+    scoring.add(IterableRule(score_per_item=0.02,
+                             iterable=job_candidate_desc_occurs,
+                             description="Occurrences between candidate full info and job description"))
+
+    scoring.add(LogicalRule(score=0.5,
+                            when=job.primary_keyword.lower() in clear_candidate_full_info,
+                            description='Job primary keyword must exist in candidate full information'))
+
+    scoring.add(SimpleRule(score=closest_date(datetime.now(timezone.utc), thread.last_updated) * DATE_WEIGTH,
+                           description=f"Considering thread date"))
+
+    if job.english_level and candidate.english_level:
+        scoring.add(SimpleRule(score=ENGLISH_WEIGTH.get(job.english_level),
```

Comment:
Здається тут помилка, не бачу щоб була евалюація чи підходить англійська кандидата по вакансії

Reply:
Справді повинно бути `candidate.english_level` замість `job.english_level`. 

Також можна зробити це так: якщо ми справді хочемо щоб під вакансію з рівнем англійської Intermediate найкраще підпадав кандидат з рівнем Intermediate, і т.д тоді варто зробити це ось так:
`scoring.add(SimpleRule(score=lower_distance(ENGLISH_WEIGTH.get(candidate.english_level), ENGLISH_WEIGTH.get(job.english_level)), description="Considering English level"))`

Я зроблю зараз коміт з другий варіантом.

Reply:
Важливо щоби рівень кандидата був не менший за вказаний у вакансії

Reply:
> Важливо щоби рівень кандидата був не менший за вказаний у вакансії

Хм, тоді краще навіть так:

`scoring.add(LogicalRule(score=ENGLISH_WEIGTH.get(candidate.english_level),
                                when=ENGLISH_WEIGTH.get(candidate.english_level) >= ENGLISH_WEIGTH.get(job.english_level),
                                otherwise=0,
                                description="Add score if candidate English level greater than job required level"))`

і трішки відкорегувати значення англійської в розрахунках:

`ENGLISH_WEIGTH = {
    EnglishLevel.NONE: 0,
    EnglishLevel.BASIC: 0.5,
    EnglishLevel.PRE: 1,
    EnglishLevel.INTERMEDIATE: 1.5,
    EnglishLevel.UPPER: 2,
    EnglishLevel.FLUENT: 2.5
}
`